src/button-with-icon/types.ts

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+
+type StyleType = "default" | "secondary" | "light" | "link";
+export interface ButtonWithIconProps
```

Comment:
to extend from `src/button/types` instead of respecifying. The only additional stuff is the icon props